code/modules/grey_market/illegalradio.dm

```diff
@@ -0,0 +1,379 @@
+#define MAIN 1
+#define SELLING 2
+#define DELIVERY 3
+
+var/list/global_illegal_radios = list()
+
+/proc/buzz_grey_market() //Updates and buzzes all illegalradios so the screen is current.
+	for(var/obj/item/device/illegalradio/radio in global_illegal_radios)
+		if(radio.notifications)
+			var/mob/living/holder = get_holder_of_type(radio, /mob/living)
+			if(holder)
+				radio.interact(holder)
+				to_chat(holder,"<span class='info'>You feel \the [radio] buzz.</span>")
+
+/obj/item/device/illegalradio/nanotrasen/New()
+	qdel(src) //Removed, but mapping issues so xd
+			
+/obj/item/device/illegalradio
+	icon = 'icons/obj/radio.dmi'
+	name = "grey market uplink"
+	suffix = "\[3\]"
+	icon_state = "illegalradio"
+	item_state = "illegalradio"
+	desc = "A modified radio with a link to the grey market. Use with caution."
+
+	siemens_coefficient = 1
+	slot_flags = SLOT_BELT
+	throw_speed = 2
+	throw_range = 9
+	w_class = W_CLASS_SMALL
+	starting_materials = list(MAT_IRON = 75, MAT_GLASS = 25)
+	w_type = RECYK_ELECTRONIC
+	melt_temperature = MELTPOINT_PLASTIC
+
+	var/list/purchase_log = list()
+	var/datum/grey_market_item/selected_item = null
+	var/datum/grey_market_player_item/new_listing = null
+	
+	var/money_stored
+	var/opened_screen = MAIN
+	var/market_cut = 0.3
+	var/new_item_fee = 25
+	var/minimum_price = 100
+	var/advanced_uplink = 0
+	var/scan_time = 150
+	var/notifications = 1 
+	var/list/sell_exceptions = list(/obj/structure/closet, /obj/item/weapon/storage)
+	
+	var/scanning = 0
+	
+/obj/item/device/illegalradio/advanced
+	name = "advanced grey market uplink"
+	icon_state = "illegalradio_adv"
+	item_state = "illegalradio_adv"
+	desc = "If there's one thing you can always trust a Vox will do, it's getting his money's worth."
+	advanced_uplink = 1
```

Comment:
you can keep this around just for having the cool alt sprite as an admin spawn/something for spessmart but I don't think this var does anything anymore?

Reply:
Nope, in the future I maybe plan on making the aforementioned "black market operative" semi-antag whose job is to make as much money as possible.